spec/base_spec.rb

```diff
@@ -76,6 +76,19 @@ def fixup_aggregate_schema(loaded_version)
     end
   end
 
+  class DummySubModel < DummyElasticSearchModel
+    attribute :my_dummy_sub_attr, :integer
+  end
+
+  class DummySubBottomLevelModel < DummySubModel; end
+  class DummySubSameLevelModel < DummyElasticSearchModel; end
```

Comment:
I'd suggest adding numbers to these sub models so it's easy to see at a glance what "layer" they represent

`DummySubModel` => `DummySub1AModel`
`DummySubBottomLevelModel` => `DummySub2AModel`
`DummySubSameLevelModel` => `DummySub1BModel`